src/sdam/server_description.ts

```diff
@@ -258,10 +258,19 @@ export function compareTopologyVersion(
   }
 
   // TODO(NODE-2674): Preserve int64 sent from MongoDB
-  const currentCounter = Long.isLong(currentTv.counter)
-    ? currentTv.counter
-    : Long.fromNumber(currentTv.counter);
-  const newCounter = Long.isLong(newTv.counter) ? newTv.counter : Long.fromNumber(newTv.counter);
+  const currentCounter =
```

Comment:
Instead of fixing it only here - should we instead fix it by adding coercion to TopologyVersion?  Right now the types are inaccurate, because TopologyVersion says counter is always a bigint.

Reply:
From slack, we have the follow-up `raw: true` issue that will take a closer look at BSON option handling. We want to prioritize unblocking `useBigInt64` usage.